### PR TITLE
RFC 0005-integer_overflow_systematic_checks

### PR DESCRIPTION
**Pull request by bsimondi**

### CODE REVIEW

**Review by ericeberry — CHANGES_REQUESTED** [review]
Please change directory to 0005-integer_overflow_systematic_checks.

Also please limit line length to 100 characters.

**Review by kihun-amlogic — APPROVED** [review]

**Review by LisShen — APPROVED** [review]

**Review by hchoi-brcm — DISMISSED** [review]

**Review by ericeberry — DISMISSED** [review]

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+- check for ((A + B) + C) + D
+Same applies for any combination of add, sub and/or mult
+
+Integer overflows sanitization checks should also apply to pointers and special care should be taking
+that the resulting pointer of an operation is within expected ranges.
+
+There are various ways of checking for integer overflows, which are documented over internet.
+Nonetheless I would encourage the code owners to look at builtins macros offered by the compilers.
+For instance, GCC offers Builtins, as documented at:
+https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html
```
Comment (mhabrat) [inline docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md:51]:
```suggestion
[GCC Integer Overflow Builtins](https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html)
```

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Some OS, like OP-TEE, offer Macros over the compilers builtins, see this is another way of increasing
+portability that you can get inspiration from, as it can be seen in
+https://github.com/OP-TEE/optee_os/blob/master/lib/libutils/ext/include/util.h
```
Comment (mhabrat) [inline docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md:54]:
```suggestion
[OP-TEE Overflow Macros](https://github.com/OP-TEE/optee_os/blob/master/lib/libutils/ext/include/util.h).
```

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+-
+
+## Future possibilities
+
+-
+
+## References
+
+-
```
Comment (mhabrat) [inline docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md:73]:
```suggestion
- [Integer Overflow or Wraparound](https://cwe.mitre.org/data/definitions/190.html).
```

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+portability that you can get inspiration from, as it can be seen in
+https://github.com/OP-TEE/optee_os/blob/master/lib/libutils/ext/include/util.h
+
+## Drawbacks
+
+-
+
+## Alternatives considered
+
+-
```
Comment (mhabrat) [inline docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md:62]:
```suggestion
- None.

```

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Open Source Dependencies
+
+-
+
+## Detailed design
+
+Systematically applies to integer operations all that involve directly or indirectly any
+field/parameter that can be under the influence of the outside world. add (A+B), sub (A-B), and mult
```
Comment (mhabrat) [inline docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md:36]:
```suggestion
field/parameter that can be under the influence of the outside world. add (A+B), subtract (A-B), and multiply
```

**Review by mhabrat — CHANGES_REQUESTED** [review]

**Review by mhabrat — CHANGES_REQUESTED** [review]
Minor nits corrected and references added.

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+bases, and is challenging for static analysers. The arithmetic behind integer overflow is well
+documented over internet. All parameters and field coming from the outside world (outside world
+being any code that calls inside of the TASECUREAPI code) could be of any value and should be
+checked against valid ranges, but that alone falls short. All resulting value of an operation
+involving one or many fields from the outside world should also be evaluated to be in within
+acceptable ranges.
+
+## Updates/Obsoletes
+
+-
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md]: Done

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+## Affected platforms
+
+-
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md]: Done

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+## Open Source Dependencies
+
+-
+
+## Detailed design
+
+Systematically applies to integer operations all that involve directly or indirectly any
+field/parameter that can be under the influence of the outside world. add (A+B), sub (A-B), and mult
+(A*B) operations are in scope. More complex operations shall be decomposed and checks for intermediate
+values, e.g. (A+B+C+B) should be decomposed on
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md]: Done

File: docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+- check independently A, B, C and D values
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply docs/rfcs/accepted/0005-integer_overflow_systematic_checks/README.md:39]: Done

**Review by ericeberry — APPROVED** [review]

**Review by ericeberry — APPROVED** [review]

**Review by mhabrat — APPROVED** [review]

**Pull request merged**